### PR TITLE
Allow source-build to remove the nuget-client patches

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Fixes: https://github.com/NuGet/Home/issues/6066
Related: https://github.com/dotnet/source-build/issues/172 & https://github.com/dotnet/source-build/pull/264
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: 

source-build folks need to a bunch of things to make NuGet build cross-plat and only CoreCLR stuff. 
They applied the following patches on our code in order to build:
https://github.com/dotnet/source-build/tree/dev/release/2.0/patches/nuget-client
A short summary of what they did:
- Don't ILMergethe Pack nupkg and only build the CoreCLR artifacts.
- Build the CLI impacting projects, by creating a "sln" and building on netstandard/netcore frameworks for those projects. 
- Do public signing xplat and delay sign on windows. 
- Bump up the MsBuild packages version. 

Through discussions with @crummel I have added incorporated these things into our repo. 

### Differences between what the patches did and what we will do now. 
- The patches built with desktop frameworks on windows NT which means different artifacts on different platforms which is not what's needed. 

### Description of the changes here:
- New Restore/Build/Pack targets.
- Add a property to tag the relevant projects for this XPLAT build. 
- Collect the relevant projects with a target call. 
- Import bootstrapped targets/props only in non-xplat (client, slnrestoremanager and tools)
- Add a switch to build only the XPLAT frameworks of the group of projects when the DotNetBuildFromSource=true param is passed. 
- Rework the Pack nupkg packing to pack with a csproj and remove the nuspec.

## Testing/Validation
Tests Added: No  
Reason for not adding tests:  Infrastructure changes
Validation done:  Manually building the source-build repo. CI build is green.

//cc 
@rrelyea

### CODE REVIEW

File: src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj [hunk 1]
```diff
@@ -4,13 +4,14 @@
   <PropertyGroup>
     <Description>NuGet wrapper for dotnet.exe</Description>
     <TargetFrameworks>netcoreapp1.0;net46</TargetFrameworks>
+    <TargetFrameworks Condition="'$(IsBuildOnlyXPLATProjects)' == 'true'">netcoreapp1.0</TargetFrameworks>
     <RuntimeIdentifier Condition=" '$(TargetFramework)' == 'net46' ">win7-x86</RuntimeIdentifier>
     <NoWarn>$(NoWarn);CS1591</NoWarn>
     <OutputType>Exe</OutputType>
-    <!-- <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">$(PackageTargetFallback);dotnet5.6</PackageTargetFallback> -->
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj:10]: 💯

File: build/common.targets [hunk 2]
```diff
@@ -142,6 +142,14 @@
     </MsBuild>
   </Target>
 
+  <Target Name="GetXPLATProject" Returns="@(XPLATProjects)">
```
Comment (emgarten) [inline build/common.targets:145]: 👍

**Review by emgarten — APPROVED** [review]
LGTM 🚀

**Comment by crummel** [comment]
Thanks for the quick turnaround on this!  The changes in approach from the patches looks good to me.  I kicked off a couple of builds with these changes and your source-build changes: [Windows](https://ci.dot.net/job/dotnet_source-build/job/dev_release_2.0/job/Windows_NT_Release_prtest/143/) and [Red Hat](https://ci.dot.net/job/dotnet_source-build/job/dev_release_2.0/job/RHEL7.2_Release_prtest/155/) and I'll keep an eye on those.

**Review by jainaashish — APPROVED** [review]
Looks good

**Comment by crummel** [comment]
Windows build had an error:
```
D:\j\workspace\Windows_NT_Re---a3f4c8f6\src\nuget-client\src\NuGet.Clients\NuGet.SolutionRestoreManager\NuGet.SolutionRestoreManager.csproj(5,3): error MSB4019: The imported project "D:\j\workspace\Windows_NT_Re---a3f4c8f6\src\nuget-client\packages\Microsoft.VSSDK.BuildTools.15.1.192\build\Microsoft.VSSDK.BuildTools.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [D:\j\workspace\Windows_NT_Re---a3f4c8f6\repos\nuget-client.proj]
D:\j\workspace\Windows_NT_Re---a3f4c8f6\src\nuget-client\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj(5,3): error MSB4019: The imported project "D:\j\workspace\Windows_NT_Re---a3f4c8f6\src\nuget-client\packages\Microsoft.VSSDK.BuildTools.15.1.192\build\Microsoft.VSSDK.BuildTools.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [D:\j\workspace\Windows_NT_Re---a3f4c8f6\repos\nuget-client.proj]
```

Does this look familiar?  I'm looking into it now.

**Comment by jainaashish** [comment]
@crummel Try `git clean -xdf` and then `configure.ps1` and build.

**Comment by nkolev92** [comment]
@crummel @jainaashish There is no configure step run there. 

I check for XPLAT, while I should be checking for build from source as well. 

Fixing it now.

**Comment by nkolev92** [comment]
@crummel 
Can you please rerun the tests now?
I've added an "AND" condition so nix should still work as before, just the windows tests need rerun.

**Comment by crummel** [comment]
Kicked it off [here](https://ci.dot.net/job/dotnet_source-build/job/dev_release_2.0/job/Windows_NT_Release_prtest/144/).

**Comment by nkolev92** [comment]
@crummel 
The build looks good. 
I will merge now.

**Comment by crummel** [comment]
Thanks!

**Pull request merged**